src/public-lib/lib/persistantApiClientCache.ts

```diff
@@ -0,0 +1,52 @@
+type NameMethodPair = {
+    name: string,
+    method: any,
+}
+
+function createCacheCheckingVersionOfGet(pair: NameMethodPair, api: any): NameMethodPair {
+    const method = function(parameters: any) {
+        // for every GET in the api, there is a corresponding 
+        // GETURL function that returns the url as a string       
+        const url = api[pair.name + "URL"]();
+        // check the cache for a matching url
+        
+        // if it exists, use that
+
+        // else, call method
+        var response =  pair.method(parameters);
+
+        // try to cache response
+
+        //return result
+        return response;
+    }
+
+    return {
+        name: pair.name,
+        method,
+    }
+}
+
+function toNameMethodPair(name: string, api: any): {name: string, method: any} {
+    return {
+        name,
+        method: api[name],
+    }
+}
+
+export function cacheGetMethods(api: any,
+                                included: string[],
+                                log?: (message: string) => void,
+                                sentryLog?: (message: string) => void
+): void {
+    //functionality check
+    if (!('caches' in self)) {
+        return;
+    }
+
+    Object.getOwnPropertyNames(api.prototype)
```

Comment:
I find this to be a confusing sequence.  I prefer the style of the existing cachePostMethod in src/public-lib/lib/apiClientCache.ts
The function is a little long, but you see that oldMethod is stored in the closure and then invoked in the new method if there isn't anything in cache